src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/CpsDiagnosticItem.cs

```diff
@@ -21,7 +19,17 @@ public CpsDiagnosticItem(CpsDiagnosticItemSource source, DiagnosticDescriptor de
         }
 
         public override ProjectId ProjectId => _source.ProjectId;
-        protected override AnalyzerReference AnalyzerReference => _source.AnalyzerReference;
+        protected override AnalyzerReference AnalyzerReference
+        {
+            get
+            {
+                // _source.AnalyzerReference can be null if the source hasn't found it's reference yet;
```

Comment:
Should we have a `Debug.Assert(_source.AnalyzerReference != null);` to ensure the contract is not violated in future?

Reply:
Added a Contract.ThrowIfNull().